generators/chipyard/src/main/scala/IOBinders.scala

```diff
@@ -0,0 +1,167 @@
+package chipyard.iobinders
+
+import chisel3._
+
+import freechips.rocketchip.config.{Field, Config, Parameters}
+import freechips.rocketchip.diplomacy.{LazyModule}
+import freechips.rocketchip.devices.debug._
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.system._
+import freechips.rocketchip.util._
+
+import sifive.blocks.devices.gpio._
+import sifive.blocks.devices.uart._
+
+import testchipip._
+import icenet._
+import tracegen.{HasTraceGenTilesModuleImp}
+
+import scala.reflect.{ClassTag, classTag}
+
+// System for instantiating binders based
+// on the scala type of the Target (_not_ its IO). This avoids needing to
+// duplicate harnesses (essentially test harnesses) for each target.
+//
+// You could just as well create a custom harness module that instantiates
+// bridges explicitly, or add methods to
+// your target traits that instantiate the bridge there (i.e., akin to
+// SimAXI4Mem). Since cake traits live in Rocket Chip it was easiest to match
+// on the types rather than change trait code.
+
+
+
+// A map of partial functions that match on the type the DUT (_not_ it's
```

Comment:
Update these comments. Feel free to remove the stuff that contrasts this against matching IO if you like.